analyzers/src/SonarAnalyzer.CSharp/Rules/TestMethodShouldContainAssertion.cs

```diff
@@ -155,7 +159,7 @@ private static bool IsTestIgnored(IMethodSymbol method)
                 .Any();
 
         private static bool IsKnownAssertion(ISymbol methodSymbol) =>
-            (KnownAssertions.GetValueOrDefault(methodSymbol.Name) is { } type && methodSymbol.ContainingType.ConstructedFrom.Is(type))
+            (KnownAssertions.GetValueOrDefault(methodSymbol.Name) is { } types && types.Any(type => methodSymbol.ContainingType.ConstructedFrom.Is(type)))
```

Comment:
Coding style nitpick
```suggestion
            (KnownAssertions.GetValueOrDefault(methodSymbol.Name) is { } types && types.Any(x => methodSymbol.ContainingType.ConstructedFrom.Is(x)))
```